src/Query/EloquentQueryBuilder.php

```diff
@@ -73,21 +92,315 @@ public function getCountForPagination()
         return $this->builder->getCountForPagination();
     }
 
+    public function limit($value)
+    {
+        $this->limit = $value;
+        $this->builder->limit($value);
+
+        return $this;
+    }
+
+    public function offset($value)
+    {
+        $this->builder->offset(max(0, $value));
+
+        if (! $this->limit) {
+            $this->limit(max(10, $value));
+        }
+
+        return $this;
+    }
```

Comment:
Where does this limit property logic come from?

Doing `$query->limit(1)->offset(1)` works already.

Reply:
Yeah that does work, but it was to get this test working - which fails without a limit as well as an offset.
https://github.com/ryanmitchell/eloquent-driver/blob/5cd764459694855d0fb2f46efd5eca92edb9dd63/tests/Data/Entries/EntryQueryBuilderTest.php#L651

Appreciate it doesnt make sense as a standalone addition, but it was an attempt at stache interoperability

Reply:
Regular Eloquent completely outside of Statamic fails when you only supply an offset too. You need both.

```php
App\Models\User::query()->offset(5)->get();
```

```
Illuminate\Database\QueryException with message 'SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'offset 5' at line 1 (SQL: select * from `users` offset 5)'
```

Reply:
No problem - thanks for clarifying what you'd expect.
I've pushed an update.